core/common/src/main/scala/net/liftweb/common/Box.scala

```diff
@@ -785,34 +822,124 @@ sealed abstract class Box[+A] extends Product with Serializable{
    * Although this function is different for true collections, because `Box` is
    * really a collection of 1, the two functions are identical.
    */
-  final def collectFirst[B](pf: PartialFunction[A, B]): Box[B] = {
+  def collectFirst[B](pf: PartialFunction[A, B]): Box[B] = {
     collect(pf)
   }
 }
 
 /**
+ * A `Full` or `Empty` box (stands in trivially for `Option`).
+ */
+sealed trait PresenceBox[+A] extends Box[A] {
+  override def isA[B](clsOrg: Class[B]): PresenceBox[B] = Empty
+  override def asA[B](implicit m: Manifest[B]): PresenceBox[B] = Empty
+
+  def or[B >: A](alternative: => Full[B]): Full[B]
+  def or[B >: A](alternative: => PresenceBox[B])(implicit a: DummyImplicit): PresenceBox[B]
+  def or[B >: A](alternative: => TryBox[B])(implicit a: DummyImplicit, b: DummyImplicit): TryBox[B]
+  def or[B >: A, E](alternative: => ParamTryBox[B, E])(implicit a: DummyImplicit, b: DummyImplicit, c: DummyImplicit): ParamTryBox[B, E]
```

Comment:
Parallel data types with implicit conversions are only an option for a small subset of these methods, because of the fact that some of the methods overlap with Iterable and such, which means if someone imports `Box._` (which is not an unreasonable thing to do in a few situations), they'll be unable to get our implicits at a higher specificity than the iterable conversions.

It ain't fun.

That said, these dummy implicits don't have anything to do with the way in which we attach these methods; instead, they are here because we're trying to overload on an parameter type with a call-by-name parameter. Overloading on parameter type is directly supported in the JVM, but by making these by-name we turn them into parameterized `Function1`s, or something roughly equivalent. That means we're suddenly overloading by the parameter's *generic* type, which isn't directly supported. Dummy implicits step in to fill this gap.